smart-contract-usage-in-javascript-sdk.md

```diff
@@ -0,0 +1,326 @@
+console.log('deployed contract address:', deployed.address);
+```
+
+
+### 5. Calling the contract functions
+
+To call the `reserve` function of the deployed contract. It can be called by anyone using the blockchain. Add after the previously added lines:
+
+```
+// call the contract 'reserve' function using the deployed address, 'sophia-address' is used to indicate we are calling using the address
+// then we pass the arguments '(200)' as count that we want to reserve and as option the amount of tokens that should be added to the contract call
+// in our case 200 (the count) times 15 (the minimum price set)
+const calledReserve = await client.contractCall(deployed.address, 'sophia-address', deployed.address, 'reserve', {
+    args: '(200)',
+    options: {amount: 200 * 15}
+}).catch(decodeError);
+
+// we log if the contract call was successful, in this case if the success response is defined
+console.log('reservation successful:', !!calledReserve);
+```
+
+To static call the `available_count` function of the deployed contract, static calls are exectued on the connected node directly and don't require a transaction or any fee, but can only be used for functions that don't manipulate state. It can be called by anyone using the blockchain. Add after the previously added lines:
+
+```
+// call the 'available_count' function without any arguments, so '()' is used
+const calledAvailableCount = await client.contractCallStatic(deployed.address, 'sophia-address', 'available_count', {args: '()'}).catch(logError);
+
+// log the bytecode result to the console
+console.log('available count bytecode:', calledAvailableCount.result);
+
+// decode the result as 'int' datatype
+const decodedAvailableCount = await client.contractDecodeData('int', calledAvailableCount.result).catch(logError);
+
+// show the available_count output on the console as int
+console.log('available count decoded:', decodedAvailableCount.value);
+```
+
+To call the `reserve` function of the deployed contract, which has to be called by the original owner, add after the previously added lines:
+
+```
+// call the 'owner_withdraw' function ithout any arguments, so '()' is used, catch any possible errors
+const calledWithdraw = await client.contractCall(deployed.address, 'sophia-address', deployed.address, 'owner_withdraw', {args: '()'}).catch(decodeError);
+
+// log the bytecode result to the console
+console.log('withdraw amount bytecode:', calledAvailableCount.result);
+
+// decode the result as 'int' datatype
+const decodedWithdraw = await client.contractDecodeData('int', calledWithdraw.result.returnValue).catch(logError);
+
+// show the withdraw amount output on the console as int
+console.log('withdraw amount decoded:', decodedWithdraw.value);
+```
+
+## Execution
+
+Add `main();` in the end of the file, then run `node index.js` to compile, deploy and call the contract. It will show output similar to:
```

Comment:
I think it would be beneficial if we run the example more frequently - after reserve, or after the count. Showing that everything works

Reply:
Still think this is valid